gammapy/modeling/models/spectral.py

```diff
@@ -1624,26 +1624,29 @@ def evaluate(self, energy, norm):
         return norm * self._evaluate((energy,), clip=True)
 
     def to_dict(self, full_output=False):
-        return {
-            self._type: {
-                "type": self.tag[0],
-                "energy": {
-                    "data": self.energy.data.tolist(),
-                    "unit": str(self.energy.unit),
-                },
-                "values": {
-                    "data": self.values.data.tolist(),
-                    "unit": str(self.values.unit),
-                },
-            }
+        data = super().to_dict(full_output)
+        data["spectral"]["energy"] = {
+            "data": self.energy.data.tolist(),
+            "unit": str(self.energy.unit),
+        }
+        data["spectral"]["values"] = {
+            "data": self.values.data.tolist(),
+            "unit": str(self.values.unit),
         }
 
+        return data
+
     @classmethod
     def from_dict(cls, data):
         data = data["spectral"]
         energy = u.Quantity(data["energy"]["data"], data["energy"]["unit"])
         values = u.Quantity(data["values"]["data"], data["values"]["unit"])
-        return cls(energy=energy, values=values)
+        model = cls(energy=energy, values=values)
```

Comment:
I suppose that `norm` should be instantiated by the `__init__`  abit like what is done for `ABLAbsorptionSpectralModel`.
See : https://github.com/gammapy/gammapy/blob/c5697fd2c91fb9728686f44d8a19bfbfd103e5be/gammapy/modeling/models/spectral.py#L1807

This would avoid modifying parameters afterwards.